controllers/argocd/secret.go

```diff
@@ -305,8 +305,14 @@ func (r *ReconcileArgoCD) reconcileExistingArgoSecret(secret *corev1.Secret, clu
 				secret.Data = make(map[string][]byte)
 			}
 
+			sessionKey, err := generateArgoServerSessionKey()
+			if err != nil {
+				return err
+			}
+
 			secret.Data[common.ArgoCDKeyAdminPassword] = []byte(hashedPassword)
 			secret.Data[common.ArgoCDKeyAdminPasswordMTime] = nowBytes()
+			secret.Data[common.ArgoCDKeyServerSecretKey] = sessionKey
```

Comment:
I believe we should only generate a new server key when the field in the Secret's data is empty. And we should not wrap it into a branch of when `hasArgoAdminPasswordChanged()` has returned true.

A little context: The server key is used for all kind of things, and should be persistent usually. It is used amongst others to sign JWTs. So if this field still exists in the Secret, it must stay the same.

Looking at this method, we should probably:
1. Move the `if secret.Data == nil` to the top of the method, out of the `hasArgoAdminPasswordChanged` condition
2. Do the same with `generateArgoServerSessionKey()`, but do this only if `secret.Data[common.ArgoCDKeyServerSecretKey] is not set to a value